concent_api/core/payments/bankster.py

```diff
@@ -71,96 +81,168 @@ def claim_deposit(
-    if settings.ADDITIONAL_VERIFICATION_COST == 0 or concent_use_case != ConcentUseCase.ADDITIONAL_VERIFICATION:
-        providers_claim_payment_info = ClaimPaymentInfo(0, 0)
-    elif available_provider_claim == 0:
-        providers_claim_payment_info = ClaimPaymentInfo(0, settings.ADDITIONAL_VERIFICATION_COST)
-    else:
-        transaction_hash = service.cover_additional_verification_cost(  # pylint: disable=no-value-for-parameter
-            provider_eth_address=provider_ethereum_address,
-            value=available_provider_claim,
-            subtask_id=subtask_id,
-        )
-        providers_claim_payment_info = ClaimPaymentInfo(
-            amount_paid=available_provider_claim,
-            amount_pending=settings.ADDITIONAL_VERIFICATION_COST - available_provider_claim,
-            tx_hash=transaction_hash,
-            payment_ts=current_time,
+        # Bankster sums the amounts of all existing DepositClaims that have the same payer as the one being processed.
+        aggregated_requestor_claims = DepositClaim.objects.filter(
+            payer_deposit_account=deposit_claim.payer_deposit_account
+        ).exclude(
+            pk=deposit_claim.pk
+        ).aggregate(
+            sum_of_existing_claims=Coalesce(Sum('amount'), 0)
         )
 
-    return requestors_claim_payment_info, providers_claim_payment_info
+        # Bankster subtracts that value from the amount of funds available in the deposit.
+        available_funds_without_claims = available_funds - aggregated_requestor_claims['sum_of_existing_claims']
+
+        # If the result is negative or zero, Bankster removes the DepositClaim object being processed.
+        if available_funds_without_claims <= 0:
+            deposit_claim.delete()
+            return None
+
+        # Otherwise if the result is lower than DepositAccount.amount,
+        # Bankster sets this field to the amount that's actually available.
+        elif available_funds_without_claims < deposit_claim.amount:
+            deposit_claim.amount = available_funds_without_claims
+
+        # If the DepositClaim still exists at this point, Bankster uses SCI to create an Ethereum transaction.
+        if deposit_claim.concent_use_case == ConcentUseCase.FORCED_ACCEPTANCE:
+            ethereum_transaction = service.force_subtask_payment(  # pylint: disable=no-value-for-parameter
+                requestor_eth_address=deposit_claim.payer_deposit_account.ethereum_address,
+                provider_eth_address=deposit_claim.payee_ethereum_address,
+                value=deposit_claim.amount,
+                subtask_id=deposit_claim.subtask_id,
+            )
+        elif deposit_claim.concent_use_case == ConcentUseCase.ADDITIONAL_VERIFICATION:
+            ethereum_transaction = service.cover_additional_verification_cost(  # pylint: disable=no-value-for-parameter
+                provider_eth_address=deposit_claim.payer_deposit_account.ethereum_address,
+                value=deposit_claim.amount,
```

Comment:
This is wrong. You're deciding whether to use `force_subtask_payment()` or `cover_additional_verification_cost()` based on the use case alone.

This means that for example in verification use case you will use `cover_additional_verification_cost()` for both requestor and the provider. Instead you should use `cover_additional_verification_cost()` to get verification fee from the provider and `force_subtask_payment()` to transfer subtask payment from the requestor to the provider.

You need to look at `deposit_claim.subtask` to know whether whether the payer is a requestor or a provider here. If that's a problem we might also add a flag to `DepositClaim` (that would require changes in the blueprint so drop me a comment if you do this).

Reply:
:+1: done